micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java

```diff
@@ -517,8 +519,8 @@ public More more() {
      * @return The state object that was passed in so the registration can be done as part
      * of an assignment statement.
      */
-    @Nullable
-    public <T> T gauge(String name, Iterable<Tag> tags, @Nullable T stateObject, ToDoubleFunction<T> valueFunction) {
+    public <T> @Nullable T gauge(String name, Iterable<Tag> tags, @Nullable T stateObject,
```

Comment:
Probably should be consistent with `Gauge` and `Function*` types. I'd vote for marking the state object as non-nullable for the reasons stated elsewhere: if the user knows the state object is nullable, they may as well not register a meter with it.